src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.AnyMobile.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.IO.IsolatedStorage
+{
+    public sealed partial class IsolatedStorageFile : IsolatedStorage, IDisposable
+    {
+        private string GetIsolatedStorageRoot()
+        {
+            return Helper.GetRootDirectory(Scope);
```

Comment:
In the `mono` implementation you linked the code appends `.isolated-storage` at the very end and I can't see it anywhere in your PR. Is that done somewhere else?

Reply:
That is done in https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs#L26 ,https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.cs#L8 
 but thanks for noticing I need to change to previous name as now it will be /IsolatedStorage/

Reply:
But the latest change will change the behavior of all platforms, not just the mobile ones, right? I'm not sure if that's something we can/should do, because it could lead to the same problem just on different platforms.

Reply:
Updated.

Reply:
How does `.config` get appended to the path on mobile if `IsolatedStorageDirectoryName` is just `.isolated-storage`?